todo-app/src/stores/TodosStore.js

```diff
@@ -0,0 +1,172 @@
+import { observable, action, runInAction } from "mobx"
+
+
+
+
+
+const createTaskFormDefault = {
+    description: "",
+    deadline: ""
+}
+
+const API_ROOT = 'https://hyf-react-api.herokuapp.com'
+
+class TodosStore {
+
+    @observable
+    todosList = []
+
+    @observable
+    createTaskForm = createTaskFormDefault
+
+    @observable
+    editingDescriptionText = ""
+
+    @observable
+    editingDeadlineTime = ""
+
+    @observable
+    taskEditingId = null
+
+    @action
+    getTodos = async () => {
+        const res = await fetch(`${API_ROOT}/todos`)
+        const parsedRes = await res.json()
+        console.log(parsedRes)
+        runInAction(() => {
+            this.todosList = parsedRes
+        })
+    }
+    @action
+    createTodo = async () => {
+        if (this.createTaskForm.description && this.createTaskForm.deadline) {
+            await fetch(`${API_ROOT}/todos/create`, {
+                method: "POST",
+                headers: {
+                    "content-type": "application/json"
+                },
+                body: JSON.stringify(this.createTaskForm)
+            })
+            runInAction(() => {
+                // reset the form
+                this.createTaskForm = createTaskFormDefault
+            })
+        } else {
+            alert("you should fill the values of Description and Deadline")
+        }
+        this.getTodos()
+    }
+
+    @action
+    removeTodo = async taskId => {
+        await fetch(`${API_ROOT}/todos/${taskId}`, {
+            method: "DELETE",
+        })
+        this.getTodos()
+
+    }
+
+    @action
+    toggleDone = async taskId => {
+        const task = this.todosList.find(task => task._id === taskId)
+        await fetch(`${API_ROOT}/todos/${taskId}`, {
+            method: "PATCH",
+            headers: {
+                'content-type': "application/json"
+            },
+            body: JSON.stringify({ done: !task.done })
+        })
+        this.getTodos()
+
+    }
+
+    @action
+    deleteAllCompleted =  () => {
```

Comment:
Looks like this action is not making an API calls. Is this still pending? A nice way to approach this would be to use `Promise.all` and make as many concurrent requests as necessary to make the action as fast as possible. It would look something like:

    await Promise.all(todosToDelete.map(todo => this.deleteTodo(todo.id)))

Assuming the `deleteTodo` function returns a `Promise`, of course!